ecommerce_api_client/client.py

```diff
@@ -1,35 +1,37 @@
 import requests
 import slumber
 
-from ecommerce_api_client.auth import JwtAuth
+from ecommerce_api_client.auth import JwtAuth, BearerAuth
 
 
 class EcommerceApiClient(slumber.API):
     DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
 
-    def __init__(self, url, signing_key, username, email, timeout=5, tracking_context=None):
+    def __init__(self, url, signing_key=None, username=None, email=None, timeout=5, tracking_context=None,
+                 oauth_access_token=None):
         """
         Instantiate a new client.
 
         Raises
-            ValueError if any of the arguments--url, signing_key, username, email--are non-truthy values.
+            ValueError if either the URL or necessary authentication values are not provided.
         """
 
-        args = (('url', url), ('signing_key', signing_key), ('username', username), ('email', email))
-        invalid_fields = []
-        for field, value in args:
-            if not value:
-                invalid_fields.append(field)
+        if not url:
+            raise ValueError('An API url must be supplied!')
 
-        if invalid_fields:
-            raise ValueError(
-                'Cannot instantiate API client. Values for the following fields are invalid: {}'.format(
-                    ', '.join(invalid_fields)))
+        if not oauth_access_token:
+            if not (signing_key and username and email):
+                raise ValueError('Either JWT or OAuth2 credentials must be suppled for authentication!')
+
+        if oauth_access_token:
+            auth = BearerAuth(oauth_access_token)
+        else:
+            auth = JwtAuth(username, email, signing_key, tracking_context)
```

Comment:
This block and the one above it could be made a bit more readable and concise as:
```python
if oauth_access_token:
    auth = BearerAuth(oauth_access_token)
elif (signing_key and username and email):
    auth = JwtAuth(username, email, signing_key, tracking_context)
else:
    raise ValueError('Either JWT or OAuth2 credentials must be suppled for authentication!')
```